### PR TITLE
Enable custom docker images

### LINKED ISSUE #295: Allow launch and exec to work with user-generated docker containers

### PR DESCRIPTION
**Pull request by davidslater**
Fixes #295

### CODE REVIEW

File: armory/__main__.py [hunk 1]
```diff
@@ -53,11 +53,12 @@ def __call__(self, parser, namespace, values, option_string=None):
...
-            raise argparse.ArgumentError(
-                self,
-                f"{values} invalid.\n"
-                f" must be one of {DOCKER_IMAGES} or {images.ALL}",
+            print(
+                f"WARNING: {values} not in "
+                f"{list(DOCKER_IMAGES.keys()) + list(DOCKER_IMAGES.values())}. "
+                "Attempting to load custom Docker image."
             )
+            setattr(namespace, self.dest, values)
```
Comment (seanpmorgan) [inline armory/__main__.py:61]: Can you add documentation to how this command would look? Perhaps in the central README:
https://github.com/twosixlabs/armory#usage

But if not maybe somewhere else in the docs.

File: docs/docker.md [hunk 2]
```diff
@@ -19,6 +19,16 @@ specified in the `docker_image` field. This enables users to extend our base ima
 and run evaluations on an image that has all additional requirements for their defense.
 
 
+### Custom Images
+If you wish to utilize custom images for armory, these can be directly specified in
+either the `docker_image` field of the config file of `armory run <path/to/config.json>`
+or in the docker image field specified in `armory launch` or `armory exec`.
```
Comment (seanpmorgan) [inline docs/docker.md:25]: What is the docker image field? Maybe write it as image CLI arg (with an example command)

File: docs/docker.md [hunk 3]
```diff
@@ -19,6 +19,16 @@ specified in the `docker_image` field. This enables users to extend our base ima
...
+
+Since Armory uses `exec_cmd` to run scenarios with already running images,
+it overrides the `CMD` of the Docker image with `tail -f /dev/null` to ensure that 
+images do not exit prematurely.
```
Comment (seanpmorgan) [inline docs/docker.md:29]: Confusing on first read. Maybe re-word to say armory executes commands on detached containers so the CMD of a docker container will be ignored or something to that effect.

File: armory/__main__.py [hunk 4]
```diff
@@ -53,11 +53,12 @@ def __call__(self, parser, namespace, values, option_string=None):
         elif values.lower() in DOCKER_IMAGES:
             setattr(namespace, self.dest, DOCKER_IMAGES[values])
         else:
-            raise argparse.ArgumentError(
-                self,
-                f"{values} invalid.\n"
-                f" must be one of {DOCKER_IMAGES} or {images.ALL}",
+            print(
```
[the remark this answers is not in the document]
Comment (davidslater) [reply armory/__main__.py:56]: I have made an issue for this: #478

File: docs/docker.md (hunk 2 above)
Comment (davidslater) [reply docs/docker.md]: Updated.

File: docs/docker.md (hunk 3 above)
Comment (davidslater) [reply docs/docker.md]: Updated.

**Comment by davidslater** [comment]
@seanpmorgan ready for another review

**Pull request merged**